controllers/placementrule/manifestwork.go

```diff
@@ -312,14 +311,26 @@ func getAllowList(client client.Client, name string) (*MetricsAllowlist, error)
 	return allowlist, nil
 }
 
-func handleDeletedMetrics(metricNameList []string) []string {
-	nameList := []string{}
-	for _, name := range metricNameList {
+func handleDeletedMetrics(defaultAllowlist []string, customAllowlist []string) []string {
+	customMetrics := []string{}
+	deletedMetrics := map[string]bool{}
+	for _, name := range customAllowlist {
 		if !strings.HasSuffix(name, "-") {
-			nameList = append(nameList, name)
+			customMetrics = append(customMetrics, name)
+		} else {
+			deletedMetrics[strings.TrimSuffix(name, "-")] = true
 		}
 	}
-	return nameList
+
+	allowMetrics := []string{}
+	for _, name := range defaultAllowlist {
+		if !deletedMetrics[name] {
+			allowMetrics = append(allowMetrics, name)
+		}
+	}
+
+	allowMetrics = append(allowMetrics, customMetrics...)
+	return allowMetrics
```

Comment:
Your current logic cannot handle this case: in custom metrics list, there are both `metrics1` and `metrics1-`. Even if user can manually remove from custom metrics list. but it is better to re-org your logic as follows:
1. form the whole metrics list w/o metrics with `-` as suffix
2. filter out metrics with `-` as suffix

Reply:
👍  cool, got it. need to handle repeated metrics.